### PR TITLE
Add Carthage support to EarlGreyExample

### PR DESCRIPTION
**Pull request by bootstraponline**
#442

I verified the Carthage tests ran successfully on both a simulator and physical iPhone.

### CODE REVIEW

**Comment by tirodkar** [comment]
LGTM

**Comment by bootstraponline** [comment]
I verified the gem is adding the copy step correctly:

carthage_script

The tests are failing to run with ` No such module 'EarlGrey'`

File: Demo/EarlGreyExample/README.md [hunk 1]
```diff
@@ -16,3 +16,14 @@ Ensure that your version of CocoaPods is upgraded to the latest 1.x version sinc
...
 Run the tests that you want to see in action.
+
+## Getting Started - Carthage
+
+From the directory containing the example source, run:
+
+- `gem install earlgrey`
+- `earlgrey install --carthage --swift-version=3.0 --target=EarlGreyExampleSwiftTests`
+- `carthage update EarlGrey --platform ios`
+- `open EarlGreyExample.xcworkspace`
```
Comment (tirodkar) [inline Demo/EarlGreyExample/README.md]: How is `EarlGreyExample.xcworkspace` generated without CocoaPods? Does it already have to exist?

Comment (bootstraponline) [inline Demo/EarlGreyExample/README.md]: Fixed, was a typo.

**Pull request merged**